src/controllers/titleController.js

```diff
@@ -10,193 +10,193 @@ const getAllTeamCodeHelper = controller.getAllTeamCodeHelper;
   const cache = cacheClosure();
 
-    const getAllTitles = function (req, res) {
-      Title.find({})
-        .then((results) => res.status(200).send(results))
-        .catch((error) => res.status(404).send(error));
+  const getAllTitles = function (req, res) {
+    Title.find({})
+      .then((results) => res.status(200).send(results))
+      .catch((error) => res.status(404).send(error));
   };
 
-    const getTitleById = function (req, res) {
-      const { titleId } = req.params;
+  const getTitleById = function (req, res) {
+    const { titleId } = req.params;
 
-      Title.findById(titleId)
-        .then((results) => res.send(results))
-        .catch((error) => res.send(error));
-    };
+    Title.findById(titleId)
+      .then((results) => res.send(results))
+      .catch((error) => res.send(error));
+  };
+
+  const postTitle = async function (req, res) {
+    const title = new Title();
+    title.titleName = req.body.titleName;
+    title.titleCode = req.body.titleCode;
+    title.teamCode = req.body.teamCode;
+    title.projectAssigned = req.body.projectAssigned;
+    title.mediaFolder = req.body.mediaFolder;
+    title.teamAssiged = req.body.teamAssiged;
+
+    const titleCodeRegex = /^[A-Za-z]+$/;
+    if (!title.titleCode || !title.titleCode.trim()) {
+      return res.status(400).send({ message: 'Title code cannot be empty.' });
+    } else if (!titleCodeRegex.test(title.titleCode)) {
+      return res.status(400).send({ message: 'Title Code must contain only upper or lower case letters.' });
+    }
 
-    const postTitle = async function (req, res) {
-      const title = new Title();
-      title.titleName = req.body.titleName;
-      title.teamCode = req.body.teamCode;
-      title.projectAssigned = req.body.projectAssigned;
-      title.mediaFolder = req.body.mediaFolder;
-      title.teamAssiged = req.body.teamAssiged;
+    // valid title name
+    if (!title.titleName.trim()) {
+      res.status(400).send({ message: 'Title cannot be empty.' });
+      return;
+    }
+
+    //  if media is empty
+    if (!title.mediaFolder.trim()) {
+      res.status(400).send({ message: 'Media folder cannot be empty.' });
+      return;
+    }
+
+    if (!title.teamCode) {
+      res.status(400).send({ message: 'Please provide a team code.' });
+      return;
+    }
+
+    const teamCodeExists = await checkTeamCodeExists(title.teamCode);
+    if (!teamCodeExists) {
+      res.status(400).send({ message: 'Invalid team code. Please provide a valid team code.' });
+      return;
+    }
+
+    // validate if project exist
+    const projectExist = await checkProjectExists(title.projectAssigned._id);
+    if (!projectExist) {
+      res.status(400).send({ message: 'Project lalala is empty or not exist!!!' });
+      return;
+    }
+
+    // validate if team exist
+    if (title.teamAssiged && title.teamAssiged._id === 'N/A') {
+      res.status(400).send({ message: 'Team not exists.' });
+      return;
+    }
+
+    title
```

Comment:
I would suggest utilizing [mongoose's built-in schema validation](https://mongoosejs.com/docs/validation.html). Then you wouldn't need to use multiple `if`s to manually validate the data.